### ISSUE TITLE
stash

### ISSUE THREAD

**Issue by smashwilson** [atom/github#557]
Create, list, and apply stashes from some kind of panel.

**Comment by cbovis** [comment]
Any thoughts on how this would fit into the UI?

**Comment by alexandernst** [comment]
The UI/UX has already been invented several times by several such applications. Just take SourceTree (or any other app that has this feature) and take a look at it.

**Comment by simurai** [comment]
Here an early mockup:

Collapsed | Expanded
--- | ---
![screen shot 2017-05-19 at 11 13 44 pm](https://cloud.githubusercontent.com/assets/378023/26251634/e92749e2-3ce8-11e7-96af-e22a798a6df6.png) | ![screen shot 2017-05-19 at 11 13 21 pm](https://cloud.githubusercontent.com/assets/378023/26251642/ed068604-3ce8-11e7-8594-e95fcfaad1ad.png)

It's part of the staging area that can be expanded/collapsed. But now that we have two lists, it might get crowded on smaller screens.

**Comment by cbovis** [comment]
@simurai the expanding list idea is slick however I agree it could get crowded as a concept if other features are implemented such as history, branches, remotes and tags. I also feel like it would be beneficial to see the changes associated with a particular stash in a similar way to how we can view unstaged changes.

Do you guys intend to eventually support features like history, tags, branches etc. in Github for Atom or is the idea to keep it lightweight pushing those responsibilities out to other plugins?

**Comment by simurai** [comment]
> @cbovis I also feel like it would be beneficial to see the changes associated with a particular stash in a similar way to how we can view unstaged changes.

You mean to show a diff in a new pane. Kinda like a preview before you actually apply the stash? Yeah, that could be handy. The diff would probably be against your current code and not how it was when you stashes something.

The ✅  in front of a stash means that it would apply cleanly without conflicts. So you can easily see which stashes are kinda out of date. Not sure if that's possible without taxing performance too much.

**Comment by cbovis** [comment]
@simurai yep thats what I mean, in the same way that you can click a file in your stage/unstaged files list to diff it against your current workspace. Seeing how the stash compares to your current workspace would make sense since that's where you would apply the stash.

**Comment by germanocorrea** [comment]
Any thoughts here? It would be really helpful stash support in Atom, and I can't find any package.

**Comment by ankurloriya** [comment]
This feature really require. Every time I need to open a terminal and do stash.

**Comment by cbovis** [comment]
@ankurloriya I'd recommend using git-plus package for stashing until this is implemented.

https://atom.io/packages/git-plus

**Comment by vprigent** [comment]
Maybe an early version could implement all those nice features in the command palette until we figure a way to have a nice GUI working for stash ?

**Comment by felops** [comment]
Any news about that? It would be awesome for my workflow..
